### PR TITLE
libdict: add new package

### PR DESCRIPTION
**Pull request by rosysong**
libdict is a C library that provides the following data structures with efficient insert, lookup, and delete routines:

  * height-balanced (AVL) tree
  * red-black tree
  * splay tree
  * weight-balanced tree
  * path-reduction tree
  * treap
  * hashtable using separate chaining
  * hashtable using open addressing with linear probing
  * skip list

All data structures in this library support insert, search, and remove, and have bidirectional iterators. The sorted data structures (everything but hash tables) support near-search operations: searching for the key greater or equal to, strictly greater than, lesser or equal to, or strictly less than, a given key. The tree data structures also support the selecting the nth element; this takes linear time, except in path-reduction and weight-balanced trees, where it only takes logarithmic time.

The API and code are written with efficiency as a primary concern. For example, an insert call returns a boolean indicating whether or not the key was already present in the dictionary (i.e. whether there was an insertion or a collision), and a pointer to the location of the associated data. Thus, an insert-or-update operation can be supported with a single traversal of the data structure. In addition, almost all recursive algorithms have been rewritten to use iteration instead.

For more details, refer to https://github.com/fmela/libdict

### CODE REVIEW

**Comment by neheb** [comment]
Does anything use this?

**Comment by rosysong** [comment]
This PR provides another choice of the requirement on dictionary library (written in C) for the developers. Currently, only libubox offer the relevant API.

**Comment by neheb** [comment]
Do you plan to submit any packages that use this?

**Comment by rosysong** [comment]
@neheb  Yes, but that package remains to be improvement.

File: libs/libdict/Makefile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+PKG_NAME:=libdict
+PKG_RELEASE=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL=https://github.com/fmela/libdict.git
+PKG_SOURCE_DATE:=2018-05-20
+PKG_SOURCE_VERSION:=95f6d27f277932272578b2554104e1e4414cfaac
+PKG_MIRROR_HASH:=6231b7d874807b333f8b1f2821f450a554c002b07d2248d04b9cfed40e2adec4
+
+PKG_LICENSE:=BSD
```
Comment (neheb) [inline libs/libdict/Makefile]: This should look like:

PKG_MAINTAINER:=Rosy Song <rosysong@rosinson.com>
PKG_LICENSE:=BSD-2-Clause
PKG_LICENSE_FILES:=LICENSE

PKG_INSTALL:=1

File: libs/libdict/Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+include $(INCLUDE_DIR)/package.mk
+
+define Package/libdict
+  SECTION:=libs
+  CATEGORY:=Libraries
+  TITLE:=C library of key-value data structures
+endef
+
+MAKE_INSTALL_FLAGS += \
+	INSTALL_USER="$(shell whoami)" \
```
Comment (neheb) [inline libs/libdict/Makefile]: I'm confused by this. Does this set the value of whoami for the host?

File: libs/libdict/Makefile [hunk 3]
```diff
@@ -0,0 +1,43 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=libdict
```
Comment (neheb) [inline libs/libdict/Makefile]: This should look like:

PKG_NAME:=libdict
PKG_SOURCE_DATE:=2018-05-20
PKG_SOURCE_VERSION:=95f6d27f277932272578b2554104e1e4414cfaac
PKG_RELEASE=1

**Comment by diizzyy** [comment]
Closing this as there are no (submitted) users of this library

**Pull request closed without merging**